### ISSUE TITLE
feat: detect cross-segment re-exports within the same slice

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#26]
## Description

The plugin currently does not detect when one segment re-exports from another segment within the same slice. This violates FSD segment boundaries.

## Example

**File:** `src/entities/cluster/model/index.ts`

```typescript
export { useClusterI18n } from "../i18n";
export type { ClusterImportanceVocab, ClusterVocab } from "../i18n";
```

Here the `model` segment re-exports from the `i18n` segment. Consumers who import from `@/entities/cluster/model` receive things that actually belong to the `i18n` segment, which creates a false dependency path and breaks segment encapsulation.

## Expected Behavior

A new rule (or extension of an existing one) should warn when a segment's public API (`model/index.ts`, `ui/index.ts`, etc.) re-exports from a sibling segment within the same slice.

For example, the correct approach would be to export `i18n` through the slice's own public API:

```typescript
// entities/cluster/index.ts (slice public API) — correct
export { useClusterI18n } from "./i18n";
export { useSelectedClusterStore } from "./model";

// entities/cluster/model/index.ts — should NOT re-export from ../i18n
```

## Why This Matters

- Blurs segment responsibilities (consumers think `useClusterI18n` is part of `model`, but it lives in `i18n`)
- Makes refactoring harder since the dependency graph does not reflect the actual structure
- Violates the principle that each segment has its own public API

**Comment by conarti** [comment]
Fixed in 2.0.0-rc.16 (200246b). You can try it with `npm i -D @conarti/eslint-plugin-feature-sliced@next`; it will ship in 2.0.0.
